src/sdk/base/BaseAccountAPI.ts

```diff
@@ -523,9 +524,12 @@ export abstract class BaseAccountAPI {
   }
 
   async signTypedData(msg: MessagePayload) {
+    const initCode = await this.getInitCode();
     return await this.services.walletService.signTypedData(
       msg,
-      this.validatorAddress
+      `0x${this.validatorAddress}`,
+      `0x${this.factoryAddress.slice(2)}`,
```

Comment:
instead of `0x${this.factoryAddress.slice(2)}` can we just write `this.factoryAddress`?

Reply:
no thats because I had the type of signTypedData and signMessage parameter to be of type Address but on BaseAccountAPI.ts its on type string so we need to type cast it one way or another somewhere